src/olympia/bandwagon/admin.py

```diff
@@ -16,6 +16,25 @@ class CollectionAddonInline(admin.TabularInline):
     # some reason, even though it's picked up as a translated fields correctly.
     readonly_fields = ('comments',)
 
+    # Permission checks:
+    # We use the implementation for CollectionAdmin permission checks: if you
+    # can edit a collection, you can edit/remove the add-ons inside it.
+    def has_change_permission(self, request, obj=None):
+        return CollectionAdmin(
+            Collection, self.admin_site).has_change_permission(
+                request, obj=getattr(obj, 'collection', None))
+
+    def has_delete_permission(self, request, obj=None):
+        # For deletion we use the *change* permission to allow curators to
+        # remove an add-on from a collection (they can't delete a collection).
+        return CollectionAdmin(
```

Comment:
nit: does it make sense to just `self.has_change_permission(request, obj)` here? (thinking DRY)

Reply:
Yeah, I debated doing that. I'll do it :)